scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -1970,44 +1951,54 @@ class Router(formatOps: FormatOps) {
     )
   }
 
-  private def shouldBreakAfterArrowInFor(
-      implicit style: ScalafmtConfig
-  ): Boolean =
-    !style.align.arrowEnumeratorGenerator && style.activeForEdition_2020_03
-
-  /**
-    * Either everything fits in one line or break on arrow.
-    *
-    * {{{
-    * xs.map {
-    *   case x =>
-    *     if (condition) doSomething
-    *     else doOtherThing
-    * }
-    * }}}
-    *
-    * in this case `expire` should be `doOtherThing` and arrow should be `=>`
-    *
-   **/
-  private def fitsOneLineOrBreakOnArrow(
-      expire: Token,
-      arrow: Token,
-      indents: Seq[Indent[Length]]
-  )(implicit style: ScalafmtConfig, line: sourcecode.Line): Seq[Split] =
-    Seq(
-      Split(Space, 0).withSingleLine(expire, killOnFail = true),
-      Split(Space, 1)
-        .withPolicy(
-          Policy(expire) {
-            case d @ Decision(t @ FormatToken(`arrow`, right, _), _)
-                // TODO(olafur) any other corner cases?
-                if !right.isInstanceOf[T.LeftBrace] &&
-                  !isAttachedSingleLineComment(t) =>
-              d.onlyNewlinesWithoutFallback
-          },
-          ignore = style.newlines.sourceIs(Newlines.fold)
-        )
-        .withIndents(indents)
-    )
+  private def getSplitsEnumerator(
+      leftFT: FormatToken
+  )(implicit style: ScalafmtConfig): Seq[Split] = {
+    val ft = nextNonCommentSameLine(leftFT)
+    val expire = lastToken(ft.meta.leftOwner)
+    val rightOwner = ft.meta.rightOwner
+    if (!style.activeForEdition_2020_03 &&
+      style.newlines.sourceIs(Newlines.classic))
+      Seq(Split(Space, 0))
+    else if (ft.right.is[T.Comment])
+      Seq(Split(Space.orNL(ft.noBreak), 0).withIndent(2, expire, After))
+    else if (style.newlines.sourceIs(Newlines.keep))
+      Seq(
+        if (ft.noBreak) Split(Space, 0)
+        else Split(Newline, 0).withIndent(2, expire, After)
+      )
+    else {
+      val spaceSplit = style.newlines.source match {
+        case _
+            if ft.right.is[T.LeftBrace] && isEnclosedInMatching(rightOwner) =>
```

Comment:
may be move it to another `else if` block above 1970 line?

Reply:
The reason I did this was because it uses the same three splits as the others below 1970. Above were the special cases.